lib/streamer/cbor_streamer.go

```diff
@@ -0,0 +1,199 @@
+package streamer
+
+import (
+	"io"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+)
+
+var _ Mux = &CborMux{}
+
+type CborMux struct {
+	// the fd tip handles append; reads all use ReadAt and track their offsets individually
+	file  *os.File
+	codec *codec.Encoder
+	wmu   sync.Mutex
+}
+
+type cborMuxRow struct {
+	Label int    `json:"l"`
+	Msg   []byte `json:"m,omitempty"`
+	Sig   int    `json:"x,omitempty"` // 1->closed
+}
+
+func CborFileMux(filePath string) Mux {
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0755)
+	if err != nil {
+		panic(errors.IOError.Wrap(err))
+	}
+	file.Write([]byte{codec.CborStreamArray})
+	return &CborMux{
+		file:  file,
+		codec: codec.NewEncoder(file, new(codec.CborHandle)),
+	}
+	// consider using runtime.`SetFinalizer to close?  currently unhandled.
+}
+
+func (m *CborMux) write(label int, msg []byte) {
+	m.wmu.Lock()
+	defer m.wmu.Unlock()
+	const magic_RAW = 0
+	const magic_UTF8 = 1
+	//	m.codec.MustEncode(cborMuxRow{
+	//		Label: label,
+	//		Msg:   msg,
+	//	})
+	_, enc := codec.GenHelperEncoder(m.codec)
+	enc.EncodeMapStart(2)
+	enc.EncodeString(magic_UTF8, "l")
+	enc.EncodeInt(int64(label))
+	enc.EncodeString(magic_UTF8, "m")
+	enc.EncodeStringBytes(magic_RAW, msg)
```

Comment:
Confirmed ([link](https://github.com/ugorji/go/issues/56))

Do we care about the order as long as it's consistent? Are you continuing to use `GenHelperEncoder` for this & the filesystem stream to avoid the developer-mentioned overhead of encoding out-of-band + compare?

I just want to make sure there's a good reason to keep using this interface.